pkg/persistence/disk_persistence_test.go

```diff
@@ -46,8 +47,40 @@ func TestDiskPersistence_Save(t *testing.T) {
 	}
 }
 
+func TestDiskPersistence_StoragePermission(t *testing.T) {
+	tempDir := "./data_storage"
+
+	err := os.Mkdir(tempDir, 000) // d---------
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		t.Fatalf("dir [%+v] was supposed to be created", tempDir)
+	}
+	defer os.RemoveAll(tempDir)
+
+	_, err = NewDiskHandle(tempDir)
+	if err == nil {
+		t.Fatalf("error was supposed to be returned")
+	}
```

Comment:
If you decide to use `reflect.DeepEqual` as Piotr suggested, I think this nil check will be redundant.

Reply:
actually, even without `reflect.DeepEqual` it feels redundant. Removed.